courses/models.py

```diff
@@ -179,7 +186,8 @@ def is_catalog_visible(self):
     @property
     def current_price(self):
         """Gets the price if it exists"""
-        product = self.products.first()
+        # breakpoint()
+        product = list(self.products.all())[0] if self.products.all() else None
```

Comment:
Commented out breakpoint can be removed.

Does calling `self.products.all())[0]` lead to a significant performance improvement? `self.products.first()` seems cleaner though I believe it does do an additional sort.

Reply:
It's just to reduce the database queries. Products are prefetched, so, trying to use that. `products` is a `GenericRelation` so that is the reason for writing it this way.